test/integration/kernel.link.spec.ts

```diff
@@ -0,0 +1,580 @@
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					slug: `relationship-card-is-xxx-forward-xxx-to-card`,
+					type: 'relationship@1.0.0',
+					name: 'is xxx-forward-xxx to',
+					data: {
+						inverseName: 'is xxx-reverse-xxx to',
+						title: 'Left',
+						inverseTitle: 'Right',
+						from: {
+							type: 'card',
+						},
+						to: {
+							type: 'card',
+						},
+					},
+				},
+			);
+
+			const linkContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					slug: `link-${contract1.slug}-is-linked-to-${contract2.slug}`,
+					type: 'link@1.0.0',
+					version: '1.0.0',
+					name: 'is xxx-reverse-xxx to',
+					data: {
+						inverseName: 'is xxx-forward-xxx to',
+						from: {
+							id: contract1.id,
+							type: contract1.type,
+						},
+						to: {
+							id: contract2.id,
+							type: contract2.type,
+						},
+					},
+				},
+			);
+
+			const element = await ctx.kernel.getContractById(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				linkContract.id,
+			);
+
+			assert(element !== null);
+		});
+
+		it('should be able to create a link from a concrete type to the any type', async () => {
+			const createContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					id: uuid(),
+					type: 'card@1.0.0',
+				},
+			);
+
+			const supportThreadContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					id: uuid(),
+					type: 'card@1.0.0',
+				},
+			);
+
+			const relationship = await ctx.kernel.replaceContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					slug: `relationship-card-forwards-to-any`,
+					type: 'relationship@1.0.0',
+					name: 'forwards to',
+					data: {
+						inverseName: 'is forwarded by',
+						title: 'Card',
+						inverseTitle: 'Destination',
+						from: {
+							type: 'card@1.0.0',
+						},
+						to: {
+							type: '*',
+						},
+					},
+				},
+			);
+
+			assert(createContract !== null);
+			assert(supportThreadContract !== null);
+			assert(relationship !== null);
+
+			const linkContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					slug: `link-${createContract.slug}-forwards-to-${supportThreadContract.slug}`,
+					type: 'link@1.0.0',
+					version: '1.0.0',
+					name: 'forwards to',
+					data: {
+						inverseName: 'is forwarded by',
+						from: {
+							id: createContract.id,
+							type: createContract.type,
+						},
+						to: {
+							id: supportThreadContract.id,
+							type: supportThreadContract.type,
+						},
+					},
+				},
+			);
+			assert(linkContract !== null);
+		});
+	});
+});
+
+async function insertCardContract() {
+	return ctx.kernel.insertContract(ctx.logContext, ctx.kernel.adminSession()!, {
+		type: 'card@1.0.0',
+	});
+}
+
+async function upsertRelationshipCardIsAttachedTo() {
```

Comment:
Could we do execute this and `upsertRelationshipIsLinkedTo()` one time in `beforeAll()` so we don't end up updating the database more than necessary when running all of the tests in CI?

Reply:
It's only executed once on `		it('should be able to create a direction-less link between two valid contracts', async () => {`